src/WebJobs.Script/Workers/ProcessManagement/WorkerProcess.cs

```diff
@@ -39,29 +42,32 @@ internal WorkerProcess(IScriptEventManager eventManager, IProcessRegistry proces
 
         public Task StartProcessAsync()
         {
-            _process = CreateWorkerProcess();
-            try
+            using (_metricsLogger.LatencyEvent(MetricEventNames.ProcessStart))
```

Comment:
Will this work in webhost scope as well? This method is called in placeholder mode to spin up worker processes.

Reply:
Yes, it does. I did the work to unify metric loggers at both scopes long back and what gets injected is this one https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script.WebHost/Diagnostics/NonDisposableMetricsLogger.cs